lib/galaxy/tools/__init__.py

```diff
@@ -1182,6 +1182,13 @@ def handle_single_execution( self, trans, rerun_remap_job_id, params, history, m
         """
         try:
             params = self.__remove_meta_properties( params )
+            if u'input' in params:
```

Comment:
Up the test failures are legitimate - looks like you are assuming something is non-None that could be None.